Jenkinsfile.deploy

```diff
@@ -98,10 +98,7 @@ pipeline {
                         sh "echo '\n-B -s ${MAVEN_SETTINGS_FILE}' | tee -a optaplanner/.mvn/maven.config"
                         if (params.MAVEN_DEPENDENCIES_REPOSITORY != '') {
                             sh "echo '\n-Denforcer.skip=true' | tee -a optaplanner/.mvn/maven.config"
-                            sh "sed -i 's|<repositories>|<repositories><repository><id>staging</id><name>Staging Repository</name><url>${params.MAVEN_DEPENDENCIES_REPOSITORY}</url><layout>default</layout><snapshots><enabled>true</enabled></snapshots><releases><enabled>true</enabled></releases></repository>|g' ${MAVEN_SETTINGS_FILE}"
-                            sh "sed -i 's|<pluginRepositories>|<pluginRepositories><pluginRepository><id>staging</id><name>Staging Repository</name><url>${params.MAVEN_DEPENDENCIES_REPOSITORY}</url><layout>default</layout><snapshots><enabled>true</enabled></snapshots><releases><enabled>true</enabled></releases></pluginRepository>|g' ${MAVEN_SETTINGS_FILE}"
-                            sh "sed -i 's|external:\\*|external:*,!staging|g' ${MAVEN_SETTINGS_FILE}"
-                            // Done to allow Maven to download release artifacts from MAVEN_DEPENDENCIES_REPOSITORY
+                            addRepositoryToSettings(params.MAVEN_DEPENDENCIES_REPOSITORY, env.MAVEN_SETTINGS_FILE)
```

Comment:
We could just store in the env under another name the path to this Settings file (`env.GLOBAL_MAVEN_SETTINGS` for example), so we don't need to do it again in the `Build[...]` part and just use the env we just stored.
The file is anyway created and stored for the rest of the pipeline run

Reply:
Truth, better than adding the repository over and over.